### ISSUE TITLE
CMake build fails with `Unknown CMake command "mbed_set_mbed_target_linker_script"`

### ISSUE THREAD

**Issue by Patater** [ARMmbed/mbed-os-example-filesystem#141]
### Description of defect

```
#### STDERR ####

#### End of STDOUT/STDERR ####

IMAGE is /builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@11/examples/mbed-os-example-devicekey/cmake_build/DISCO_L475VG_IOT01A/develop/GCC_ARM/mbed-os-example-devicekey.bin
[EXAMPLES]> INFO     In folder 'mbed-os-example-thread-statistics'
[EXAMPLES]> INFO     In folder 'mbed-os-example-sys-info'
[EXAMPLES]> INFO     In folder 'mbed-os-example-cpu-stats'
[EXAMPLES]> INFO     In folder 'mbed-os-example-socket-stats'
[EXAMPLES]> INFO     In folder 'mbed-os-example-error-handling'
[EXAMPLES]> INFO     In folder 'mbed-os-example-filesystem'
[EXAMPLES]> INFO     Compiling mbed-os-example-filesystem DISCO_L475VG_IOT01A GCC_ARM
[EXAMPLES]> INFO     Executing command 'mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c'...

#### STDOUT ####
Configuring project and generating build system...
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.7 (found version "3.7.4") found components: Interpreter 
-- Checking for Python package prettytable -- found
-- Checking for Python package future -- found
-- Checking for Python package jinja2 -- found
-- Checking for Python package intelhex -- found
-- Configuring incomplete, errors occurred!
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@11/examples/mbed-os-example-filesystem/cmake_build/DISCO_L475VG_IOT01A/develop/GCC_ARM/CMakeFiles/CMakeOutput.log".
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@11/examples/mbed-os-example-filesystem/cmake_build/DISCO_L475VG_IOT01A/develop/GCC_ARM/CMakeFiles/CMakeError.log".

#### STDERR ####
CMake Error at CMakeLists.txt:18 (mbed_set_mbed_target_linker_script):
  Unknown CMake command "mbed_set_mbed_target_linker_script".
```

#### Target(s) affected by this defect ?
DISCO_L475VG_IOT01A

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM

#### What version of Mbed-os are you using (tag or sha) ?

9bbc8c914c7e 

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
Mbed CLI 2

#### How is this defect reproduced ?
`mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c`

**Comment by ciarmcom** [comment]
Thank you for raising this detailed GitHub issue. I am now notifying our internal issue triagers. 
*Internal Jira reference: https://jira.arm.com/browse/IOTOSM-3786*

**Comment by LDong-Arm** [comment]
PR created: https://github.com/ARMmbed/mbed-os-example-filesystem/pull/142

**Comment by LDong-Arm** [comment]
Fix merged.

**Comment by 0xc0170** [comment]
Fixed via #141 (it was not automatically closed?).  I was about to close it @LDong-Arm  was faster :)
